lib/plugins/aws/package/compile/events/cognitoUserPool/index.test.js

```diff
@@ -264,6 +264,53 @@ describe('AwsCompileCognitoUserPoolEvents', () => {
       ).to.equal('AWS::Lambda::Permission');
     });
 
+    it('should be overrideable from Resources', () => {
+      awsCompileCognitoUserPoolEvents.serverless.service.functions = {
+        first: {
+          events: [
+            {
+              cognitoUserPool: {
+                pool: 'MyUserPool',
+                trigger: 'PreSignUp',
+              },
+            },
+          ],
+        },
+      };
+      awsCompileCognitoUserPoolEvents.serverless.service.resources = {
+        CognitoUserPoolMyUserPool: {
+          Type: 'AWS::Cognito::UserPool',
+          Properties: {
+            UserPoolName: 'ProdMyUserPool',
+            MfaConfiguration: 'OFF',
+            EmailVerificationSubject: 'Your verification code',
+            EmailVerificationMessage: 'Your verification code is {####}.',
+            SmsVerificationMessage: 'Your verification code is {####}.',
+          },
+        },
+      };
+
+
+      awsCompileCognitoUserPoolEvents.compileCognitoUserPoolEvents();
+
+      expect(awsCompileCognitoUserPoolEvents.serverless.service.provider
+        .compiledCloudFormationTemplate.Resources
+        .CognitoUserPoolMyUserPool.Type
+      ).to.equal('AWS::Cognito::UserPool');
+      expect(Object.keys(awsCompileCognitoUserPoolEvents.serverless.service.provider
```

Comment:
The length checks can be simplified - to sound more natural:
```
expect(_.keys(awsCompileCognitoUserPoolEvents.serverless.service.provider
        .compiledCloudFormationTemplate.Resources
        .CognitoUserPoolMyUserPool.Properties))
    .to.have.lengthOf(6);
```

Additionally, lodash should be preferred over the native functions, to ensure compatibility over multiple Node versions that could be used by the user (e.g. `_.keys()`, `_.assign()`, etc.)